### PR TITLE
[AMORO-1872] Support operating the Paimon table in Terminal

### LINKED ISSUE #1872: [Subtask]: Support operating the Paimon table in Terminal.
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### Description

Support operating the Paimon table in Terminal.

### Parent issue

#1269 

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by HuangFru**
## Why are the changes needed?

Close #1872.

## Brief change log

- Add the paimon-spark dependency.
- Fix the "dispose" method in TerminalManager is never used.
- Add the Paimon catalog configuration in sparkConf.
- Refactor some code to make it more clear.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (**yes**)
- If yes, how is the feature documented? (***not applicable***)

### CODE REVIEW

File: ams/server/pom.xml [hunk 1]
```diff
@@ -164,6 +164,12 @@
             </exclusions>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.paimon</groupId>
+            <artifactId>paimon-spark-common</artifactId>
+            <version>${paimon.version}</version>
```
Comment (baiyangtx) [inline ams/server/pom.xml:170]: please add as scope:runtime

**Review by baiyangtx — COMMENTED** [review]
Cloud you please add some screen shot?

File: ams/server/pom.xml (hunk 1 above)
Comment (HuangFru) [reply ams/server/pom.xml:170]: down

Comment (HuangFru) [reply ams/server/pom.xml:170]: done

**Comment by HuangFru** [comment]
> Cloud you please add some screen shot?

Sure, I'll add it later.

**Comment by shidayang** [comment]
Have you tested in a Kerberos environment?

**Comment by HuangFru** [comment]
> Have you tested in a Kerberos environment?

Yes, both Simple and Kerberos have been tested.

**Comment by shidayang** [comment]
> > Have you tested in a Kerberos environment?
> 
> Yes, both Simple and Kerberos have been tested.

Very good

**Comment by HuangFru** [comment]
This is a Paimon catalog with Kerberos:
POPO20231013-143441
And there are some tables in it(also created by the terminal):
image
Executed 'select' SQL in the Terminal:
image
Due to some Iceberg and Mix formats' code being modified, test Mix-Iceberg:
image
Iceberg:
image

**Review by shidayang — APPROVED** [review]

**Review by baiyangtx — APPROVED** [review]
LGTM, thanks for your contributions

**Comment by shidayang** [comment]
LGTM

**Pull request merged**